Mobile v15 metadata component

lauty95 opened the pull request:
Related issue #161 

Doubts:
Source and Receiving are mocked from stonecrop, is it ok?

"Components" and "Quantity" are fixed, if I use component's slot from BeamMetadata component, I should move "ControlButtons" from Transfer, something like:

```javascript
<ControlButtons
    :onCreate="create"
    :onSubmit="() => store.submit<StockEntry>('Stock Entry', stockEntryId)"
    :onCancel="() => store.cancel<StockEntry>('Stock Entry', stockEntryId)" />

<BeamMetadata :order="workOrder">
    <template #components>
        <ListView :items="items" />
    </template>
</BeamMetadata>
```


![Captura de pantalla 2024-10-09 a la(s) 10 53 34 a  m](https://github.com/user-attachments/assets/d46c163c-986e-49ab-a724-90fff77c295c)

lauty95 committed:
fix: data validation

lauty95 committed:
feat: metadata on WO

lauty95 committed:
chore: yarn

lauty95 committed:
Merge branch 'mobile_v15' into mobile_v15-metadata-component

agritheory commented:
@lauty95 Can you rebase this to `mobile_V15_se_workflow branch` and incorporate the appropriate components from https://github.com/agritheory/stonecrop/pull/171? Let @crabinak know if you need a hand

agritheory committed:
Merge branch 'mobile_v15' into mobile_v15-metadata-component

lauty95 committed:
feat: supplier added on sqlite

lauty95 committed:
Merge branch 'mobile_v15' into mobile_v15-metadata-component

lauty95 committed:
feat: recive context

lauty95 committed:
feat: customer and supplier

lauty95 committed:
fix: print

lauty95 committed:
fix: rejected_qty

lauty95 committed:
fix: stock_qty

lauty95 commented:
Receive context:
Added supplier and warehouse to description

![Captura de pantalla 2024-11-20 a la(s) 10 05 00 a  m](https://github.com/user-attachments/assets/f9021333-6f72-4aff-9f7a-ca5cc136852b)

Ship context:
Added warehouse and customer when it comes from a sale order

![Captura de pantalla 2024-11-20 a la(s) 10 07 35 a  m](https://github.com/user-attachments/assets/f6b0415a-270d-482e-9fb5-fa45d26400f5)

Transfer context: WIP

I worked in sqlite to add customer and supplier fields
On ship context change onMounted to useInfiniteScroll

agritheory committed:
chore: linters
